api/src/main/scala/Geoprocessing.scala

```diff
@@ -0,0 +1,65 @@
+package org.wikiwatershed.mmw.geoprocessing
+
+import java.util.concurrent.atomic.LongAdder
+
+import collection.concurrent.TrieMap
+
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+
+import geotrellis.raster._
+import geotrellis.raster.rasterize._
+import geotrellis.vector._
+import geotrellis.vector.io._
+
+import geotrellis.spark._
+
+trait Geoprocessing extends Utils {
+  /**
+    * For an InputJson object, return a histogram of raster grouped count results.
+    *
+    * @param   input  The InputData
+    * @return         A histogram of results in JSON format
+    */
+  def getRasterGroupedCount(input: InputData) = {
+    val aoi = createAOIFromInput(input)
+    val rasterLayers = createRasterLayerIds(input.rasters, input.zoom, aoi)
+    Map("result" -> rasterGroupedCount(rasterLayers, aoi)).toJson
+  }
+
+  /**
+    * From a sequence of rasterLayers and a shape, return a list of pixel counts.
+    *
+    * @param   rasterLayers  The key (e.g. "input.rasterCRS")
+    * @param   multiPolygon  The AOI as a MultiPolygon
+    * @return                A Map of cell counts
+    */
+  private def rasterGroupedCount(
+    rasterLayers: Seq[TileLayerCollection[SpatialKey]],
+    multiPolygon: MultiPolygon
+  ): Map[String, Int] = {
+    val init = () => new LongAdder
+    val update = (_: LongAdder).increment()
+    // assume all the layouts are the same
```

Comment:
Turns out this was the wrong assumption (https://github.com/WikiWatershed/model-my-watershed/issues/2153) *but* I think MMW may be passing in the nlcd layer as the first raster in all the multi-layer grouped count operations (https://github.com/WikiWatershed/model-my-watershed/blob/31c5712d8c7f0477080fd862d8e6c38358c7a8ef/src/mmw/mmw/settings/base.py) which would standardize things.